src/restic/backend/gs/config.go

```diff
@@ -0,0 +1,71 @@
+package gs
+
+import (
+	"path"
+	"strings"
+
+	"restic/errors"
+	"restic/options"
+)
+
+// Config contains all configuration necessary to connect to an gcs compatible
+// server.
+type Config struct {
+	Location    string
+	ProjectID   string
+	JSONKeyPath string
+	Bucket      string
+	Prefix      string
+	Layout      string `option:"layout" help:"use this backend layout (default: auto-detect)"`
+
+	Connections uint `option:"connections" help:"set a limit for the number of concurrent connections (default: 20)"`
+}
+
+// NewConfig returns a new Config with the default values filled in.
+func NewConfig() Config {
+	return Config{
+		Connections: 20,
+	}
+}
+
+func init() {
+	options.Register("gs", Config{})
+}
+
+const defaultPrefix = "restic"
```

Comment:
Drop the default prefix, that's also a deprecated setting of the `s3` backend. If no path is given in the location, assume it is `/`.